### PR TITLE
GH-31: Add type definition for commander

### PR DESCRIPTION
**Pull request by qtran-n4**
Task: https://github.com/NumberFour/n4jsd/issues/31

### CODE REVIEW

File: npm/commander/2.19.0/@n4jsd/commander/index.n4jsd [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @see https://github.com/tj/commander.js/
+ */
+export external public class __Command {
+    public version(v: string, customFlag: string=): __Command;
+    public arguments(args: string): __Command
+    public description(desc: string): __Command;
+    public usage(usg: string): __Command
+    public option(opt: string, descr: string, fn: Function | string=): __Command;
```
Comment (dbo) [inline npm/commander/2.19.0/@n4jsd/commander/index.n4jsd]: You can also also use a `RegExp`, and additionally define a default, and the defaults may be any value, not just strings (e.g. integers), so I think it's:

`public option(opt: string, descr: string, collector: RegExp|Function| any=, default: any=): Command;`

I know `any` in a union hardly makes sense, but the union gives some notion of how to use the API.

File: npm/commander/2.19.0/@n4jsd/commander/index.n4jsd [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @see https://github.com/tj/commander.js/
+ */
+export external public class __Command {
```
Comment (dbo) [inline npm/commander/2.19.0/@n4jsd/commander/index.n4jsd]: I think this should be `class Command` right away.

File: npm/commander/2.19.0/@n4jsd/commander/index.n4jsd [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    public command(cmd: string, descr: string, config: ~Object with {isDefault: boolean}): __Command;
+    public alias(alias: string): __Command;
+    public action(fn: Function): __Command;
+    public parse(argv: [string]): Object;
+    public args: [string];
+    public help: Function;
+}
+// By default, export an instance of Command
+export default external public const program: __Command;
+export external public const Command: constructor{__Command};
```
Comment (dbo) [inline npm/commander/2.19.0/@n4jsd/commander/index.n4jsd]: So you don't need this, if you export the class above.

**Review by dbo — CHANGES_REQUESTED** [review]

**Review by torstenkraemer — APPROVED** [review]
I think this is OK, but in general I'missing these links in the definition, either formal or informal in a description: (Only hot guesses, don't know if they are right)
- https://www.npmjs.com/package/commander
- https://github.com/tj/commander.js (Ok, that is in the Jsdoc)

Also: Which version is targeted?   I assume latest as of today but who can verify that next week?

File: npm/commander/2.19.0/@n4jsd/commander/index.n4jsd (hunk 2 above)
Comment (qtran-n4) [reply npm/commander/2.19.0/@n4jsd/commander/index.n4jsd]: done

File: npm/commander/2.19.0/@n4jsd/commander/index.n4jsd (hunk 3 above)
Comment (qtran-n4) [reply npm/commander/2.19.0/@n4jsd/commander/index.n4jsd]: done

**Comment by qtran-n4** [comment]
@dbo: Can you have a second look?

File: npm/commander/2.19.0/@n4jsd/commander/index.n4jsd [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    public arguments(args: string): Command
+    public description(desc: string): Command;
+    public usage(usg: string): Command
+    public option(opt: string, descr: string, collector: Function | RegExp | any=,deft: any=): Command;
+    public command(cmd: string, descr: string, config: ~Object with {isDefault: boolean}): Command;
+    public alias(alias: string): Command;
+    public action(fn: Function): Command;
+    public parse(argv: [string]): Object;
+    public args: [string];
+    public help: Function;
```
Comment (dbo) [inline npm/commander/2.19.0/@n4jsd/commander/index.n4jsd]: define a proper method `help(): void`

**Review by dbo — APPROVED** [review]
with that changed

File: npm/commander/2.19.0/@n4jsd/commander/index.n4jsd (hunk 4 above)
Comment (qtran-n4) [reply npm/commander/2.19.0/@n4jsd/commander/index.n4jsd]: done

**Comment by qtran-n4** [comment]
Thanks!

**Pull request merged**